x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorCustomSchedule.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class ConnectorCustomSchedule implements Writeable, ToXContentObject {
+
+    private final Map<String, Object> configurationOverrides;
+    private final boolean enabled;
+    private final String interval;
+    private final String lastSynced;
+    private final String name;
+
+    private ConnectorCustomSchedule(
+        Map<String, Object> configurationOverrides,
```

Comment:
What's the data we anticipate storing in overrides? Is it structured enough to warrant something more granular than `Map<String,Object>`?

Reply:
This `configurationOverrides` can override anything related to crawler settings. The fact that crawler is mixed up with Connector doc is a result of an old project that aimed to represent crawler as a connector (it's not done yet). `configurationOverrides ` can be a pretty complex object and I would prefer to avoid going down the rabbit hole to represent this correctly here. What do you think, is this approach acceptable?

Reply:
Actually, I took a deeper look into documentation and example docs and this object seems feasible to implement: https://www.elastic.co/guide/en/enterprise-search/8.9/crawler-managing.html#crawler-managing-multiple-schedules will add its representation :+1: